src/Laravel/ApiPlatformProvider.php

```diff
@@ -1090,15 +1090,18 @@ public function boot(ResourceNameCollectionFactoryInterface $resourceNameCollect
 
         $this->loadViewsFrom(__DIR__.'/resources/views', 'api-platform');
 
-        $fieldsBuilder = $this->app->make(FieldsBuilderEnumInterface::class);
-        $typeBuilder = $this->app->make(ContextAwareTypeBuilderInterface::class);
-        $typeBuilder->setFieldsBuilderLocator(new ServiceLocator(['api_platform.graphql.fields_builder' => $fieldsBuilder]));
+        $config = $this->app['config'];
+
+        if ($config->get('api-platform.graphql.enabled')) {
+            $fieldsBuilder = $this->app->make(FieldsBuilderEnumInterface::class);
+            $typeBuilder = $this->app->make(ContextAwareTypeBuilderInterface::class);
+            $typeBuilder->setFieldsBuilderLocator(new ServiceLocator(['api_platform.graphql.fields_builder' => $fieldsBuilder]));
+        }
```

Comment:
Is this intended @soyuka? it looks like `$fieldsBuilder`is unused. It's maybe a remaining of something?

Reply:
we're in the `boot` method, this is to handle a circular dependency between the FieldsBuilder and theTypeBuilder of graphql...